redwood-protocol-compose/src/commonMain/kotlin/app/cash/redwood/protocol/compose/ProtocolBridge.kt

```diff
@@ -15,84 +15,44 @@
  */
 package app.cash.redwood.protocol.compose
 
-import app.cash.redwood.protocol.ChildrenDiff
 import app.cash.redwood.protocol.ChildrenTag
 import app.cash.redwood.protocol.Diff
 import app.cash.redwood.protocol.Event
 import app.cash.redwood.protocol.EventSink
 import app.cash.redwood.protocol.Id
-import app.cash.redwood.protocol.LayoutModifiers
-import app.cash.redwood.protocol.PropertyDiff
 import app.cash.redwood.widget.Widget
+import kotlinx.serialization.json.Json
 
-public class ProtocolBridge : EventSink {
-  private var nextValue = Id.Root.value + 1L
-  private val nodes = mutableMapOf<Id, DiffProducingWidget>()
-
-  private var childrenDiffs = mutableListOf<ChildrenDiff>()
-  private var layoutModifiers = mutableListOf<LayoutModifiers>()
-  private var propertyDiffs = mutableListOf<PropertyDiff>()
-
-  public fun nextId(): Id {
-    val value = nextValue
-    nextValue = value + 1L
-    return Id(value)
-  }
-
-  public fun append(childrenDiff: ChildrenDiff) {
-    childrenDiffs += childrenDiff
-  }
-
-  public fun append(layoutModifiers: LayoutModifiers) {
-    this.layoutModifiers += layoutModifiers
-  }
-
-  public fun append(propertyDiff: PropertyDiff) {
-    propertyDiffs += propertyDiff
-  }
-
+/**
+ * Exposes a [Widget.Children] and [Widget.Provider] whose changes can be captured as a [Diff] to
+ * send to a remote frontend. Incoming [Event]s can also be sent to this instance and will be routed
+ * to the appropriate handler.
+ */
+public interface ProtocolBridge : EventSink {
```

Comment:
This is an entirely-new type unrelated to the old one aside from sharing the `EventSink` + `createDiffOrNull` APIs.